core/shared/src/main/scala/com/lynbrookrobotics/potassium/streams/ZippedStream.scala

```diff
@@ -0,0 +1,36 @@
+package com.lynbrookrobotics.potassium.streams
+
+class ZippedStream[A, B](aPeriodicity: ExpectedPeriodicity,
+                         bPeriodicity: ExpectedPeriodicity) extends Stream[(A, B)] {
+  override val expectedPeriodicity: ExpectedPeriodicity = (aPeriodicity, bPeriodicity) match {
+    case (Periodic(a), Periodic(b)) =>
+      if (a eq b) {
+        Periodic(a)
+      } else {
+        NonPeriodic
+      }
+
+    case _ => NonPeriodic
+  }
+
+  private[this] var aSlot: Option[A] = None
+  private[this] var bSlot: Option[B] = None
+
+  def maybePublishPair(): Unit = {
+    if (aSlot.isDefined && bSlot.isDefined) {
+      publishValue((aSlot.get, bSlot.get))
+      aSlot = None
+      bSlot = None
+    }
+  }
+
+  def receiveA(aValue: A): Unit = {
+    aSlot = Some(aValue)
+    maybePublishPair()
+  }
+
+  def receiveB(bValue: B): Unit = {
+    bSlot = Some(bValue)
+    maybePublishPair()
+  }
+}
```

Comment:
Does not have strong reference to parent streams?
data.zip(otherData.map(...))
will break stuff cause there is are no strong references to otherData.map(...)